### PR TITLE
Handle null properties

### PR DESCRIPTION
**Pull request by devkat**
* Avoid NPE in collection conversion when model.getProperties() returns null.
* Use case objects instead of case classes when parameter list is empty.
* Convert swagger names to Scala identifiers.

### CODE REVIEW

File: lib/src/main/scala/eu/unicredit/swagger/generators/DefaultModelGenerator.scala [hunk 1]
```diff
@@ -31,7 +31,11 @@ class DefaultModelGenerator extends ModelGenerator with SwaggerConversion {
 
     val params: Iterable[ValDef] = for ((pname, prop) <- props) yield PARAM(pname, propType(prop)): ValDef
 
-    val tree: Tree = CLASSDEF(GenClass) withFlags Flags.CASE withParams params
+    val tree: Tree =
+      if (params.isEmpty)
+        OBJECTDEF(GenClass) withFlags Flags.CASE
```
Comment (andreaTP) [inline lib/src/main/scala/eu/unicredit/swagger/generators/DefaultModelGenerator.scala:51]: that is good indeed!

File: build.sbt [hunk 2]
```diff
@@ -17,7 +17,10 @@ lazy val lib = project.
     name := """sbt-swagger-codegen-lib""",
     libraryDependencies ++= Seq(
       "com.eed3si9n" %% "treehugger" % "0.4.1",
-      "io.swagger" % "swagger-parser" % "1.0.23"
+      "io.swagger" % "swagger-parser" % "1.0.23",
+      "com.google.guava" % "guava" % "20.0",
```
Comment (andreaTP) [inline build.sbt]: guava is not needed

File: build.sbt [hunk 3]
```diff
@@ -17,7 +17,10 @@ lazy val lib = project.
...
+      "org.scalactic" %% "scalactic" % "3.0.1",
```
Comment (andreaTP) [inline build.sbt]: scalactic is not yet used

File: lib/src/test/scala/eu/unicredit/swagger/ScalaUtilsSpec.scala [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package eu.unicredit.swagger
+
+import ScalaUtils._
+import org.scalatest._
+import Matchers._
+
+class ScalaUtilsSpec extends FlatSpec {
```
Comment (andreaTP) [inline lib/src/test/scala/eu/unicredit/swagger/ScalaUtilsSpec.scala:7]: We really need some testing, thanks a lot for starting this :-)

**Review by andreaTP — CHANGES_REQUESTED** [review]

**Comment by andreaTP** [comment]
thanks a lot for this contribution.

File: build.sbt (hunk 2 above)
Comment (devkat) [reply build.sbt]: It is used in ScalaUtils for the case format conversion.

Comment (devkat) [reply build.sbt]: Never mind, I saw that it is already included transitively.

**Comment by andreaTP** [comment]
LGTM
Great, can you squash commits please? and than I will merge!

**Comment by fralken** [comment]
Hello, this is a nice improvement, thanks! But there are still some problems. It is nice to have names converted to Scala identifiers, but right now this works only for the model. Both client and server generated code is broken, because it uses unconverted names. Have you time to check this?

btw, the last commit is broken (it contains unmerged code).

Anyway, thanks for this pr!

**Comment by devkat** [comment]
Thanks for reviewing the PR. Unfortunately we have cancelled the project in question and I cannot invest any more time. Feel free to use any of the code changes.

**Comment by fralken** [comment]
Ok, I cherry picked some of your commits. Thanks anyway.

**Pull request closed without merging**